examples/generative/dreambooth.py

```diff
@@ -0,0 +1,640 @@
+preservation" refer to [this document](https://dreambooth.github.io/).
+
+Here, we need to introduce a few key terms specific to DreamBooth:
+
+* **Unique class**: Examples include "dog", "person", etc. In this example, we use "dog". 
+* **Unique identifier**: A unique identifier that is prepended to the unique class while
+forming the "instance prompts". In this example, we use "sks" as this unique identifier. 
+* **Instance prompt**: Denotes a prompt that best describes the "instance images". An
+example prompt could be - "f"a photo of {unique_id} {unique_class}". So, for our example,
+this becomes - "a photo  of sks dog".
+* **Class prompt**: Denotes a prompt without the unique identifier. This prompt is used
+for generating "class images" for prior preservation. For our example, this prompt is -
+"a photo of dog".
+* **Instance images**: Denote the images that represent the visual concept you're trying
+to teach aka the "instance prompt". This number is typically just 3 - 5. We typically
+gather these images ourselves. 
+* **Class images**: Denote the images generated using the "class prompt" for using prior
+preservation in DreamBooth training. We leverage the pre-trained model before fine-tuning
+it to generate these class images. Typically, 200 - 300 class images are enough.
+
+In code, this generation process looks quite simply:
+
+```py
+from tqdm import tqdm
+import numpy as np 
+import hashlib
+import keras_cv
+import PIL 
+import os
+
+class_images_dir = "class-images"
+os.makedirs(class_images_dir, exist_ok=True)
+
+model = keras_cv.models.StableDiffusion(img_width=512, img_height=512, jit_compile=True)
+
+class_prompt = "a photo of dog"
+num_imgs_to_generate = 200 
+for i in tqdm(range(num_imgs_to_generate)):
+    images = model.text_to_image(
+        class_prompt,
+        batch_size=3,
+    )
+    idx = np.random.choice(len(images))
+    selected_image = PIL.Image.fromarray(images[idx])
+    
+    hash_image = hashlib.sha1(selected_image.tobytes()).hexdigest()
+    image_filename = os.path.join(class_images_dir, f"{hash_image}.jpg")
+    selected_image.save(image_filename)
+```
+
+To keep the runtime of this example short, the authors of this example have gone ahead
+and generated some class images using
+[this notebook](https://colab.research.google.com/gist/sayakpaul/6b5de345d29cf5860f84b6d04d958692/generate_class_priors.ipynb). 
+
+**Note** that prior preservation is an optional technique used in DreamBooth, but it
+almost always helps in improving the quality of the generated images. 
+"""
+
+instance_images_root = tf.keras.utils.get_file(
+    origin="https://huggingface.co/datasets/sayakpaul/sample-datasets/resolve/main/instance-images.tar.gz",
+    untar=True,
+)
+class_images_root = tf.keras.utils.get_file(
+    origin="https://huggingface.co/datasets/sayakpaul/sample-datasets/resolve/main/class-images.tar.gz",
+    untar=True,
+)
+
+"""
+## Visualize images
```

Comment:
Note to self:

* Add some explanations on how the instance images should be constructed in general (different poses, and so on). 
* Also visualize the instance images. 
* Add a simpler method to visualize.